### ISSUE TITLE
Added shops with the "Shops" overlay quest seems to be ignored in the star count

### ISSUE THREAD

**Issue by n-peugnet** [streetcomplete/StreetComplete#5074]
Similarly to what is explained in [this FAQ entry](https://wiki.openstreetmap.org/wiki/StreetComplete/FAQ#Why_does_the_star_counter_not_match_the_sum_of_previous_stars_and_upload_counter) and this issue https://github.com/streetcomplete/StreetComplete/issues/1165, my star counter by type for Shops dropped from 144 to 47 once I stopped editing. Unfortunately I didn't take a screenshot when it was at 144.

This seems strange as I don't see how any conflict can occur with newly created nodes, and they are not notes.

Here is my changeset that seems to have been completely ignored: https://www.openstreetmap.org/changeset/137331223

**How to Reproduce**

I haven't exactly tested this but it should produce the described behaviour:

1. Add shops whithin the "Shops" layer
2. See the counters increase. 
3. After some time they jump back to their previous value.

**Expected Behavior**

The counter should not decrease as there should be no conflicts.

**Versions affected**

v53.1

<details>
<summary><strong>Screenshots</strong></summary>

![Screenshot_20230614-192730](https://github.com/streetcomplete/StreetComplete/assets/23519418/0506aeed-01c6-440c-95fb-01963722d3a0)
![Screenshot_20230614-192810](https://github.com/streetcomplete/StreetComplete/assets/23519418/2f8ab842-77db-4abb-8731-4370ceeae930)

</details>

**Comment by westnordost** [comment]
Did they jump back to 47 immediately after upload, or some time after upload?

**Comment by n-peugnet** [comment]
Some time after upload, around 20 minutes I would say. I was already marked as "synchronized" (like in the second screenshot) when I had 144 stars. I have 4G and auto-upload always enabled.

**Comment by westnordost** [comment]
Oh, there it is...: https://github.com/streetcomplete/sc-statistics-service/blob/master/classes/ChangesetAnalyzer.class.php#L44

**Comment by n-peugnet** [comment]
> Alright, then it seems it may rather be a problem of the backend

This is what I thought too but as I was not sure it was safer to just post it to the app repo. Feel free to move it in the backend repo.

**Comment by westnordost** [comment]
Hmm, the logic has been intricate to best 1:1 map the changeset data from what would count as "one edit" in StreetComplete. 

For example, a split-way should count only as one edit in StreetComplete but creates up to as many nodes as the user splits the way at plus creates as many ways as the user splits the way. With the linked logic, it has been possible to look for that.
Also, undoing (reverting) an edit should count as -1 edits (+1 - 1 = 0) in StreetComplete, but on OpenStreetMap, it is 2 edits.

The logic has been written before it was possible to add nodes (through overlays), delete nodes ("does not exist" answer) and create vertices (e.g. address overlay) and is thus incorrect now.

While it may be possible to introduce more logic to the existing logic in order to correctly detect the creation of nodes as edits (but not the creation of nodes as part of splitting a way), what will not be possible is to do the same for **adding vertices** (e.g. adding an entrance with address in the addresses overlay). So, given that it is not possible to achieve 100% correctness and that if one was to abuse the scoring, it would be easier to do this with the overlays anyway, I think I will just use the OpenStreetMap edit count, i.e. not process the data further.

### In a nutshell

I think I will change the statistics aggregator backend to just count the OpenStreetMap edits and not do some expensive analyzing on it. (Last time I cleared the data, it took 3 months or so to re-aggregate the data). This will lead to users having a slightly higher star count after the statistics are synced than offline, especially if they have to split ways often or revert their changes often.

**Reasons**:
- it's much less processing (and code) expensive to do this
- I believe the chance of abuse is low - not higher than what can be done with the overlays
- some low percentage of more (or less) star count doesn't really matter that much, IMO

**Comment by n-peugnet** [comment]
> - some low percentage of more (or less) star count doesn't really matter that much, IMO

I agree with this, also if it is small enough it should not be as noticeable.
In my case I must admit that it was quite depressing to see the work of two hours not being counted correctly. I do not use SC only for the star count but it seeing it go down quite a lot discouraged me from using it for some days.

**Comment by westnordost** [comment]
Alright, I made the changes. I cleared the statistics database so that they are aggregated anew. Last time it took 3 months. 

This time, it will probably take around 3 days.

**Comment by matkoniecz** [comment]
> I believe the chance of abuse is low - not higher than what can be done with the overlays

And if someone is making deliberately bad edits with StreetComplete: please ask them to stop via changeset comment and if they are not stopping, please write to Data Working Group ( https://wiki.openstreetmap.org/wiki/Data_working_group ), like with any other vandalism.

(with systematic mistakes due to StreetComplete interface: create a new issue)
